Winleafs-Server/Winleafs.Server.Services/SpotifyService.cs

```diff
@@ -0,0 +1,181 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using SpotifyAPI.Web;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Winleafs.Server.Models.Models;
+using Winleafs.Server.Services.DTO;
+using Winleafs.Server.Services.Exceptions;
+using Winleafs.Server.Services.Helpers;
+using Winleafs.Server.Services.Interfaces;
+
+namespace Winleafs.Server.Services
+{
+    public class SpotifyService : ISpotifyService
+    {
+        private static HttpClient _spotifyAuthorizationClient = new HttpClient();
+
+        private IUserService _userService;
+        private DbContext _context;
+
+        public SpotifyService(IUserService userService, DbContext context)
+        {
+            _userService = userService;
+            _context = context;
+        }
+
+        public async Task SwapToken(string code, string applicationId)
+        {
+            var user = await _userService.FindUserByApplicationId(applicationId);
+
+            if (user == null)
+            {
+                throw new InvalidApplicationIdException("No user was found for the given application id");
+            }
+
+            _spotifyAuthorizationClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", GetSpotifyAuthorizationHeader());
+
+            var postDictionary = new Dictionary<string, string>
```

Comment:
Class variable instead of method as its a one time initialized thing

Reply:
Moved it to constructor, even better